resources/metric.mjs

```diff
@@ -0,0 +1,90 @@
+import * as Statistics from "./statistics.mjs";
+
+/** Number of milliseconds in a single minute. */
+export const MILLIS_PER_MIN = 60 * 1000;
+
+export class Metric {
+    /** @param {string} name */
+    constructor(name, unit = "ms") {
+        if (typeof name !== "string")
+            throw new Error(`Invalid metric.name=${name}, expected string.`);
+        this.name = name;
+        this.unit = unit;
+
+        this.mean = 0.0;
+        this.geomean = 0.0;
+        this.delta = 0.0;
+        this.percentDelta = 0.0;
+
+        this.sum = 0.0;
+        this.min = 0.0;
+        this.max = 0.0;
+
+        /** @type {number[]} */
+        this.values = [];
+
+        /** @type {Metric} */
+        this.parent = undefined;
+        /** @type {Metric[]} */
+        this.children = [];
+
+        // Mark properties which refer to other Metric objects as
+        // non-enumerable to avoid issue with JSON.stringify due to circular
+        // references.
+        Object.defineProperties(this, {
+            parent: {
+                writable: true,
+            },
+            children: {
+                writable: true,
+            },
+        });
+    }
+
+    get shortName() {
+        return this.parent ? this.name.replace(this.parent.name + "-", "") : this.name;
+    }
+
+    get valueString() {
+        const mean = this.mean.toFixed(2);
+        if (!this.percentDelta || !this.delta)
+            return `${mean} ${this.unit}`;
+        return `${mean} ± ${this.delta.toFixed(2)} (${this.percentDelta.toFixed(1)}%) ${this.unit}`;
+    }
+
+    get length() {
+        return this.values.length;
+    }
+
+    /** @param {Metric} metric */
+    addChild(metric) {
+        if (metric.parent)
+            throw new Error("Cannot re-add sub metric");
+        metric.parent = this;
+        this.children.push(metric);
+    }
+
+    /** @param {number} value */
+    add(value) {
+        if (typeof value !== "number")
+            throw new Error(`Adding invalid value=${value} to metric=${this.name}`);
+        this.values.push(value);
+    }
+
+    /**
+     * Calculate aggregate metrics for collected values.
+     */
+    compute() {
```

Comment:
Please call this function computeAggregatedMetrics or something and remove the comment.

Reply:
DONE.